src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp

```diff
@@ -721,38 +714,76 @@ void SF45LaserSerial::sf45_process_replies(float *distance_m)
 
 				break;
 
-			case 2:
+			case ROTATION_RIGHT_FACING:
 				scaled_yaw = scaled_yaw + 90; // rotation facing right
 				break;
 
-			case 3:
+			case ROTATION_LEFT_FACING:
 				scaled_yaw = scaled_yaw - 90; // rotation facing left
 				break;
 
 			default:
 				break;
 			}
 
-			// Convert to meters for rangefinder update
+			// Convert to meters for the debug message
 			*distance_m = raw_distance * SF45_SCALE_FACTOR;
-			obstacle_dist_cm = (uint16_t)raw_distance;
+			_current_bin_dist = ((uint16_t)raw_distance < _current_bin_dist) ? (uint16_t)raw_distance : _current_bin_dist;
 
 			uint8_t current_bin = sf45_convert_angle(scaled_yaw);
 
-			// If we have moved to a new bin
-
 			if (current_bin != _previous_bin) {
+				PX4_DEBUG("scaled_yaw: \t %d, \t current_bin: \t %d, \t distance: \t %8.4f\n", scaled_yaw, current_bin,
+					  (double)*distance_m);
+
+				if (_current_bin_dist > _obstacle_map_msg.max_distance) {
+					_current_bin_dist = _obstacle_map_msg.max_distance + 1; // As per ObstacleDistance.msg definition
+				}
+
+				// if the sensor has its cycle delay configured for a low value like 5, it can happen that not every bin gets a measurement.
+				// in this case we assume the measurement to be valid for all bins between the previous and the current bin. win
+				uint8_t start;
+				uint8_t end;
+
+				if (abs(current_bin - _previous_bin) > BIN_COUNT / 4) {	// wrap-around case
```

Comment:
The BIN_COUNT / 4 is just a threshold to determine if we have flipped from one side of the array to the other? E.g. we were on bin 0 and now we are on bin 71 since the lidar passed the center line?

Reply:
Yea, if they are further apart than BIN_COUNT/4, then i assume we passed the zero line. 
There might be better ways to implement this, but thats the easyest i thought of.